### PR TITLE
Fix page redirection bug and issues in the silent sign-in flow

### PR DESCRIPTION
**Pull request by thivi**
- Resolves issues when using the `trySignInSilently` flow.
- Resolves the issue of code getting executed even after a page redirection is triggered using `location.href =` statement. 
- Sets a timeout to expire a `trySignInSilently` request.

### CODE REVIEW

**Review by inthirakumaaran — APPROVED** [review]

**Review by DimalChandrasiri — APPROVED** [review]

File: lib/src/clients/main-thread-client.ts [hunk 1]
```diff
@@ -565,20 +569,26 @@ export const MainThreadClient = async (
...
                         .then((response: BasicUserInfo) => {
                             window.removeEventListener("message", listenToPromptNoneIFrame);
+                            clearTimeout(timer);
                             resolve(response);
                         })
                         .catch((error) => {
                             window.removeEventListener("message", listenToPromptNoneIFrame);
+                            clearTimeout(timer);
                             reject(error);
                         });
```
Comment (brionmario) [inline lib/src/clients/main-thread-client.ts:593]: Shall we do `clearTimeout(timer);` in finally block.

File: lib/src/clients/main-thread-client.ts [hunk 2]
```diff
@@ -297,14 +297,14 @@ export const MainThreadClient = async (
     ): Promise<BasicUserInfo> => {
         const config = await _dataLayer.getConfigData();
 
-        const isLoggingOut = await _sessionManagementHelper.receivePromptNoneResponse(
+        const shouldStopContinue = await _sessionManagementHelper.receivePromptNoneResponse(
```
Comment (brionmario) [inline lib/src/clients/main-thread-client.ts:300]: const shouldStopContinue: boolean =

Comment (thivi) [reply lib/src/clients/main-thread-client.ts:300]: Resolved by https://github.com/asgardeo/asgardeo-auth-spa-sdk/pull/51

File: lib/src/clients/main-thread-client.ts (hunk 1 above)
Comment (thivi) [reply lib/src/clients/main-thread-client.ts:593]: Resolved by https://github.com/asgardeo/asgardeo-auth-spa-sdk/pull/51

**Pull request merged**